plugin/pkg/auth/authenticator/token/bootstrap/bootstrap.go

```diff
@@ -184,3 +192,28 @@ func parseToken(s string) (string, string, error) {
 	}
 	return split[1], split[2], nil
 }
+
+// getGroups loads and validates the bootstrapapi.BootstrapTokenGroupsKey key
+// from the bootstrap token secret, returning a list of group names or an error
+// if any of the group names are invalid.
+func getGroups(secret *api.Secret) ([]string, error) {
+	groupsString := getSecretString(secret, bootstrapapi.BootstrapTokenGroupsKey)
+	if groupsString == "" {
+		return []string{bootstrapapi.BootstrapDefaultGroup}, nil
+	}
+
+	// split on commas amd validate the prefix on each group name
+	groups := strings.Split(groupsString, ",")
+	for _, group := range groups {
+		if group == bootstrapapi.BootstrapDefaultGroup {
+			// allow the default group even though it doesn't strictly match the prefix
+			continue
+		}
+
+		if !strings.HasPrefix(group, bootstrapapi.BootstrapGroupPrefix) {
```

Comment:
I didn't want to add any restrictions that might be surprising, but I don't think some basic restrictions would be that hard to work with. Maybe something like `\Asystem:bootstrappers:[a-zA-Z0-9:-]{1,255}\z`? 

Is it okay to just put a little `ValidateBootstrapGroupName(name string) error` function into `pkg/bootstrap/api/validate.go` alongside the constants in `types.go`? Exporting the regex as a `const` feels weirder to me than just exporting a function.